lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -965,7 +965,7 @@ namespace Js
         JavascriptPromiseAsyncSpawnStepArgumentExecutorFunction* asyncSpawnStepArgumentExecutorFunction = JavascriptPromiseAsyncSpawnStepArgumentExecutorFunction::FromVar(function);
         Var argument = asyncSpawnStepArgumentExecutorFunction->GetArgument();
 
-        JavascriptFunction* next = JavascriptFunction::FromVar(JavascriptOperators::GetProperty(asyncSpawnStepArgumentExecutorFunction->GetGenerator(), PropertyIds::next, function->GetScriptContext()));
+        JavascriptFunction* next = function->GetScriptContext()->GetLibrary()->EnsureGeneratorNextFunction();
```

Comment:
Just occurred to me that this is more than an optimization, it fixes a bug where the user could have changed async method behavior by changing the generator prototype next property.

Glad we are making this change to the implementation.